### ISSUE TITLE
FALSE NEGATIVE: `LI_LAZY_INIT_STATIC` misses lazy initialization from a method call

### ISSUE THREAD

**Issue by rikki322** [spotbugs/spotbugs#4276]
Hi, I found a false negative in SpotBugs 4.10.3 when a static field is lazily initialized with the result of a method call.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs LI_LAZY_INIT_STATIC`

**Minimal reproducer**

```java
class SpotbugsSonarqubeLazyInitMethodCall {
  static String value;

  String test(StringBuilder builder) {
    if (value == null) value = builder.toString();
    return value;
  }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-sonarqube-li-lazy-init-static-fn-method-call.java
spotbugs -textui -effort:max -low -visitors LazyInit classes
```

**Current behavior**

SpotBugs produces no `LI_LAZY_INIT_STATIC` diagnostic.

**Expected behavior**

SpotBugs should report `LI_LAZY_INIT_STATIC` at line 5 because the unsynchronized method lazily initializes the static field `value`.
